amoro-ams/src/main/java/org/apache/amoro/server/dashboard/OverviewCache.java

```diff
@@ -0,0 +1,296 @@
+
+import org.apache.amoro.metrics.Counter;
+import org.apache.amoro.metrics.Gauge;
+import org.apache.amoro.metrics.Metric;
+import org.apache.amoro.metrics.MetricDefine;
+import org.apache.amoro.metrics.MetricKey;
+import org.apache.amoro.metrics.MetricSet;
+import org.apache.amoro.server.dashboard.model.OverviewDataSizeItem;
+import org.apache.amoro.server.dashboard.model.OverviewResourceUsageItem;
+import org.apache.amoro.server.dashboard.model.OverviewTopTableItem;
+import org.apache.amoro.shade.guava32.com.google.common.collect.ImmutableList;
+import org.apache.amoro.shade.guava32.com.google.common.collect.Maps;
+import org.apache.amoro.shade.guava32.com.google.common.collect.Sets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class OverviewCache {
+
+  public static final String STATUS_PENDING = "Pending";
+  public static final String STATUS_PLANING = "Planing";
+  public static final String STATUS_EXECUTING = "Executing";
+  public static final String STATUS_IDLE = "Idle";
+  public static final String STATUS_COMMITTING = "Committing";
+
+  private static final Logger log = LoggerFactory.getLogger(OverviewCache.class);
+  private static volatile OverviewCache INSTANCE;
+
+  private Map<MetricKey, Metric> registeredMetrics;
+  private Map<MetricDefine, List<MetricKey>> metricDefineMap;
+  private Map<String, Long> optimizingStatusCountMap = new ConcurrentHashMap<>();
+  private ConcurrentLinkedDeque<OverviewResourceUsageItem> resourceUsageHistory =
+      new ConcurrentLinkedDeque<>();
+  private ConcurrentLinkedDeque<OverviewDataSizeItem> dataSizeHistory =
+      new ConcurrentLinkedDeque<>();
+
+  private volatile List<OverviewTopTableItem> allTopTableItem = new ArrayList<>();
+
+  private AtomicInteger totalCatalog = new AtomicInteger();
+  private AtomicLong totalDataSize = new AtomicLong();
+  private AtomicInteger totalTableCount = new AtomicInteger();
+  private AtomicInteger totalCpu = new AtomicInteger();
+  private AtomicLong totalMemory = new AtomicLong();
+  private int maxRecordCount;
+
+  private OverviewCache() {}
+
+  /** @return Get the singleton object. */
+  public static OverviewCache getInstance() {
+    if (INSTANCE == null) {
+      synchronized (OverviewCache.class) {
+        if (INSTANCE == null) {
+          INSTANCE = new OverviewCache();
+        }
+      }
+    }
+    return INSTANCE;
+  }
+
+  public void initialize(int maxRecordCount, MetricSet globalMetricSet) {
+    this.maxRecordCount = maxRecordCount;
+    this.registeredMetrics = globalMetricSet.getMetrics();
```

Comment:
How about storing the global metric set rather than the metric map?
The global metric set is a singleton implementation.
However, there is no guarantee that the metric map inside will not change during implementation, even though it seems to be fine at the moment.